services/pseudo_controller/src/main/java/com/futurewei/alcor/pseudo_controller/pseudo_controller.java

```diff
@@ -0,0 +1,313 @@
+
+        // setting neighbor state of port 1 on node 2
+
+        // add a new neighbor state with CREATE
+        Neighbor.NeighborState.Builder new_neighborState_builder_port_2 = Neighbor.NeighborState.newBuilder();
+        new_neighborState_builder_port_2.setOperationType(Common.OperationType.CREATE);
+
+        // fill in neighbor state structs of port 3
+        Neighbor.NeighborConfiguration.Builder NeighborConfiguration_builder_node_2 = Neighbor.NeighborConfiguration.newBuilder();
+        NeighborConfiguration_builder_node_2.setRevisionNumber(2);
+        NeighborConfiguration_builder_node_2.setVpcId(vpc_id_1);
+        NeighborConfiguration_builder_node_2.setId(port_id_1);
+        NeighborConfiguration_builder_node_2.setMacAddress("6c:dd:ee:00:00:02");
+        NeighborConfiguration_builder_node_2.setHostIpAddress(aca_node_one_ip);
+
+        Neighbor.NeighborConfiguration.FixedIp.Builder neighbor_fixed_ip_builder_node_2 = Neighbor.NeighborConfiguration.FixedIp.newBuilder();
+        neighbor_fixed_ip_builder_node_2.setNeighborType(Neighbor.NeighborType.L2);
+        neighbor_fixed_ip_builder_node_2.setSubnetId(subnet_id_1);
+        neighbor_fixed_ip_builder_node_2.setIpAddress("10.0.0.2");
+
+        NeighborConfiguration_builder_node_2.addFixedIps(neighbor_fixed_ip_builder_node_2.build());
+
+        new_neighborState_builder_port_2.setConfiguration(NeighborConfiguration_builder_node_2.build());
+        Neighbor.NeighborState neighborState_two = new_neighborState_builder_port_2.build();
+        GoalState_builder.putNeighborStates(neighborState_two.getConfiguration().getId(), neighborState_two);
+        Goalstate.ResourceIdType resource_id_type_neighbor_two = Goalstate.ResourceIdType.newBuilder()
+                .setType(Common.ResourceType.NEIGHBOR).setId(neighborState_two.getConfiguration().getId()).build();
+        host_resource_builder_node_two.addResources(resource_id_type_neighbor_two);
+//        Goalstate.HostResources host_resources_neighbor_two = Goalstate.HostResources.newBuilder()
+//                .addResources(resource_id_type_neighbor_two).build();
+//        GoalState_builder.putHostResources(aca_node_two_ip, host_resources_neighbor_two);
+
+        // end of setting neighbor state of port 1 on node 2
+
+
+        // end of setting up port 2 on aca node 2
+        GoalState_builder.putHostResources(aca_node_one_ip, host_resource_builder_node_one.build());
+        GoalState_builder.putHostResources(aca_node_two_ip, host_resource_builder_node_two.build());
+        Goalstate.GoalStateV2 message = GoalState_builder.build();
+        System.out.println("Built GoalState successfully, GoalStateV2 content: \n"+message.toString()+"\n");
+        System.out.println("Host resources count: " + message.getHostResourcesCount());
+        System.out.println("Time to call the GRPC functions");
+
+//        ManagedChannel channel = ManagedChannelBuilder.forAddress(aca_node_one_ip, 123).usePlaintext().build();
```

Comment:
aca_node_one_ip should change to NCM IP right, since we are sending the goal state v2 to NCM

Reply:
You're right. 

In my setup, aca is running on the same node with aca, that's why I'm setting it to the IP of aca node one. 

Will make the grpc address a param so it is more flexible.